semgrep/general.yaml

```diff
@@ -0,0 +1,15 @@
+rules:
+- id: verbose-booleans
+  patterns:
+    - pattern-either: 
+      - pattern: if ($EXPR) { return true; } else { return false; }
+      - pattern: if (!$EXPR) { return false; } else { return true; }
+      - pattern: if ($EXPR) { return true; } return false;
+      - pattern: if (!$EXPR) { return false; } return true;
+  message: returning exhaustive booleans should just return the expression instead
+  metadata: {
+      todo: "as of 0.29.0, the autofix sometimes works, and sometimes goes haywire"
+      }
+  languages: [ts]
+  severity: ERROR
+  fix: return $EXPR;
```

Comment:
This was a fascinating exercise--all four bits of code in the matchers can be replaced with this simple fix. But wait, it is actually even better! To semgrep, these are the same (as they should be):
```
if ($EXPR) { return true; } else { return false; }
if ($EXPR) return true; else return false;
```
And, if you are wondering, line breaks are completely transparent, too.